### PR TITLE
Griffin 1.0.0

### PR DESCRIPTION
**Pull request by guoyuepeng**

### CODE REVIEW

File: griffin-core/src/main/java/org/apache/griffin/core/worker/schedule/TaskDispatcherScheduler.java [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * 任务执行调度期 和 dispatcher交互
```
Comment (toyboxman) [inline griffin-core/src/main/java/org/apache/griffin/core/worker/schedule/TaskDispatcherScheduler.java:26]: there're some comments using Chinese, we should translate them and align same style with other parts

File: griffin-anomalydetection/pom.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.apache.griffin</groupId>
+        <artifactId>griffin-parent</artifactId>
+        <version>${revision}</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>griffin-anomalydetection</artifactId>
```
Comment (toyboxman) [inline griffin-anomalydetection/pom.xml:30]: anomalydetection looks a bit long name, coud you find a simple name for it?

File: pom.xml [hunk 3]
```diff
@@ -97,22 +112,61 @@ under the License.
...
+            <artifactId>lombok</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <!-- Test dependencies -->
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+<!--        <dependency>-->
```
[the remark this answers is not in the document]
Comment (guoyuepeng) [reply pom.xml:163]: add it here for later usage.

[the remark this answers is not in the document]
Comment (guoyuepeng) [reply pom.xml:163]: add it here for later usage.

**Comment by guoyuepeng** [comment]
we need to refactor our scheduler so that we might leverage customer's existing scheduler engines.
most of time, most customer maintain one scheduler engine.

**Comment by guoyuepeng** [comment]
existing trigger are usually like

public interface Trigger {
    @Nullable
    Date nextExecutionTime(TriggerContext triggerContext);
}

but we need something like

Trigger with CitiesDimension with SeasonDimension


those cities are dynamically injected

**Comment by toyboxman** [comment]
LGTM.

Let's merge ASAP, then all paticipants of the commnity can continue to contribute and bring out the new grffin.

Thanks

**Comment by guoyuepeng** [comment]
> LGTM.
> 
> Let's merge ASAP, then all paticipants of the commnity can continue to contribute and bring out the new grffin.
> 
> Thanks

a long way to go, let's make it faster.

**Comment by toyboxman** [comment]
Hello, @guoyuepeng 

Could you merge this, let's start with latest basis.

**Pull request closed without merging**